### ISSUE TITLE
Using linter-coffeelint with coffee-react-transform

### ISSUE THREAD

**Issue by hongrich** [AtomLinter/linter-coffeelint#25]
I'm using `coffee-react-transform` with coffeelint. I have the following in my `coffeelint.json` file

```
  "coffeelint": {
    "transforms": [ "coffee-react-transform" ]
  }
```

This used to work before the most recent upgrade, but now I get the following error:

```
Error: Cannot find module 'coffee-react-transform'
  at Module._resolveFilename (module.js:351:15)
  at Function.Module._resolveFilename (/Applications/Atom.app/Contents/Resources/app/src/module-cache.js:383:52)
  at Function.Module._load (module.js:293:25)
  at Module.require (module.js:380:17)
  at require (module.js:399:17)
  at Object.module.exports.require (/Users/rich/.atom/packages/linter-coffeelint/node_modules/coffeelint/lib/ruleLoader.js:23:14)
  at Object.coffeelint.lint (/Users/rich/.atom/packages/linter-coffeelint/node_modules/coffeelint/lib/coffeelint.js:266:30)
  at LinterCoffeelint.lintFile (/Users/rich/.atom/packages/linter-coffeelint/lib/linter-coffeelint.coffee:49:25)
  at /Users/rich/.atom/packages/linter/lib/linter-view.coffee:158:18
  at Array.forEach (native)
  at /Users/rich/.atom/packages/linter/lib/linter-view.coffee:157:18
  at FSReqWrap.oncomplete (fs.js:100:15)
```

**Comment by AsaAyers** [comment]
It looks like it needs to be installed as part of your project. Try 

    npm install --save-dev coffee-react-transform

As of v0.2.0 linter-coffeelint will `require()` coffeelint instead of trying to run the CLI and parse the output. Atom packages can't require npm packages that were installed with `npm install -g`.

Please close if that fixes it, if not let me know and I'll continue investigating.

**Comment by hongrich** [comment]
It was already part of my project as a devDependencies
```
  "devDependencies": {
    "coffee-react-transform": "^2.4.1"
  },
```

I also attempted to re-install it again, and that doesn't seem to make any difference. Thanks again for looking into this.

**Comment by AsaAyers** [comment]
This is a bug in CoffeeLint. The `ruleLoader` assumes that `process.cwd()` is going to be the root of your project, but that isn't true here. I also maintain CoffeeLint, so I'll get that fixed. Right after I figure out the right solution.

I created a project of my own for verifying this. As a temporary workaround if you start atom from the command line from inside your project it will work. If you work with multiple projects in atom at once you may be stuck though. All windows share the same `process.cwd()`, so I had to close all of my atom windows and launch it again from the project that uses `coffee-react-transform`.

**Comment by hongrich** [comment]
Oh cool. Thanks for tracking down this bug. I can confirm that the workaround does work for now.

Definitely looking forward to the real fix. Thanks!

**Comment by rylanc** [comment]
This may be an atom issue, but this doesn't seem to be working for me. Whenever I launch atom from my project folder, my `process.cwd()` is set to `/`. If I change the working dir (`process.chdir()`), the linter starts working. Any idea why the CWD would be wrong?

**Comment by AsaAyers** [comment]
From what I've seen (using Linux) cwd is inherited from the first place you launch Atom. If you go into your project and launch atom from there it will match that project. I work on multiple projects at once and all Atom windows share the same `process.cwd()`, so if I go to one of my other projects and launch atom the cwd will be "wrong" from the perspective of that second project.

**Comment by AsaAyers** [comment]
I fixed this in CoffeLint v1.9.5. If your project uses a local coffeelint (devDependency in your package.json) you will need to update that and upgrade linter-coffeelint to v0.2.3.

linter-coffeelint < 0.2.3 has a bug that causes it to not be able to find the local coffeelint https://github.com/AtomLinter/linter-coffeelint/commit/7ffb6eb9466569debfd5933218165bef8c0e2aef.

**Comment by hongrich** [comment]
Thanks @AsaAyers. Just updated and seems to work! 👍
